syntax = "proto3";

service SimulationEventReceiver {
    rpc NotifyEvent(NotifyEventRequest) returns (NotifyEventResponse);
    rpc SyncCommand(SyncCommandRequest) returns (SyncCommandResponse);
}

message NotifyEventRequest {
    int32 event_id = 1;
    string event_name = 2;
    map<string, string> event_args = 3;
    int32 command_id = 4;
}

message NotifyEventResponse {
    bool success = 1;
}

message SyncCommandRequest {
}

message SyncCommandResponse {
    bool has_command = 1;
    int32 command_id = 2;
    string command_name = 3;
    map<string, string> command_args = 4;
}